src/python/pants/testutil/rule_runner.py

```diff
@@ -135,6 +135,11 @@ def engine_error(
                 f"expected: {contains}\n\n"
                 f"exception: {underlying}"
             )
+    else:
+        raise AssertionError(
+            "DID NOT RAISE ExecutionError with underlying exception type "
+            f"{expected_underlying_exception}."
+        )
```

Comment:
@Eric-Arellano I completely missed this in the review of #13108, I guess I got lost during one of the major refactorings in that PR..